implementations/rust/ockam/ockam_entity/src/profile_set.rs

```diff
@@ -11,62 +11,125 @@ pub use identifiers::*;
 pub mod secrets;
 pub use secrets::*;
 
+use crate::EntityError::{InvalidInternalState, InvalidParameter};
+use ockam_core::hashbrown::hash_map::HashMap;
+
 /// An Entity represents an identity in various authentication contexts.
 #[derive(Clone)]
-pub struct Entity<P: ProfileTrait> {
+pub struct ProfileSet<P: ProfileTrait> {
     default_profile_identifier: ProfileIdentifier,
-    profiles: Vec<P>,
+    profiles: HashMap<ProfileIdentifier, P>,
+}
+
+pub trait ProfileRetrieve<P: ProfileTrait> {
+    fn profile(&self, profile_identifier: &ProfileIdentifier) -> Option<&P>;
```

Comment:
I think because we'll be adding that to EntityWorker, return type should be Option<P> (or maybe Result<P>, or Result<Option<P>>)

Reply:
We don't want to move the `Profile` out of the `ProfileSet` (soon to be in Entity itself), do we? The profiles are stored in a `HashMap` by identifier. Typically we would `get` a reference to a contained value. Would we need to pull it out mutably? Since it's a worker, do we need to?

Reply:
I found use cases for both, but the signatures need more work.